src/core/async-work-cache.ts

```diff
@@ -50,19 +64,21 @@ export class AsyncWorkCache<K, V> {
     return promise;
   }
 
-  delete (key: K) {
-    this._keyToResultMap.delete(key);
-  }
-
-  clear() {
-    this._keyToResultMap.clear();
-  }
-
-  set(key: K, value: V) {
-    this._keyToResultMap.set(key, Promise.resolve(value));
+  /**
+   * Returns a copy of the underlying map which can be mutated as necessary
+   * and then used to create another cache.
+   */
+  fork(): Map<K, Promise<V>> {
```

Comment:
I'm not sure fork is the right name for this, considering it returns a different type.